primitives/merkle-mountain-range/src/lib.rs

```diff
@@ -351,6 +351,37 @@ impl_leaf_data_for_tuple!(A:0, B:1, C:2);
 impl_leaf_data_for_tuple!(A:0, B:1, C:2, D:3);
 impl_leaf_data_for_tuple!(A:0, B:1, C:2, D:3, E:4);
 
+/// A MMR proof data for a group of leaves.
+#[derive(codec::Encode, codec::Decode, RuntimeDebug, Clone, PartialEq, Eq)]
+pub struct BatchProof<Hash> {
+	/// The indices of the leaves the proof is for.
+	pub leaf_indices: Vec<LeafIndex>,
+	/// Number of leaves in MMR, when the proof was generated.
+	pub leaf_count: NodeIndex,
+	/// Proof elements (hashes of siblings of inner nodes on the path to the leaf).
+	pub items: Vec<Hash>,
+}
+
+impl<Hash> From<BatchProof<Hash>> for Proof<Hash> {
+	/// Will panic if leaf indices is less than 1
+	fn from(proof: BatchProof<Hash>) -> Self {
+		Proof {
+			leaf_index: proof.leaf_indices[0],
+			leaf_count: proof.leaf_count,
+			items: proof.items,
+		}
+	}
+}
```

Comment:
I'm not sure about this.. it's a slippery slope to implicitly enable these types of lossy conversions.

What do you think of replacing this with a custom converter function with a suggestive name like `into_first_proof()`/`into_single_leaf_proof()` that needs to be explicitly called instead of implicit conversions using `trait From`?